h/schemas/base.py

```diff
@@ -99,6 +100,61 @@ def serialize(self, node, appstruct):
     return EnumType
 
 
+def _colander_exception_msg(exc):
+    """
+    Combine error messages from a `colander.Invalid` exception.
+
+    :type exc: colander.Invalid
+    :rtype str:
+    """
+    msg_dict = exc.asdict()
+    for child in exc.children:
+        msg_dict.update(child.asdict())
+    msg_list = ["{}: {}".format(field, err) for field, err in msg_dict.items()]
+    return "\n".join(msg_list)
+
+
+def validate_query_params(schema, params):
+    """
+    Validate query parameters using a Colander schema.
+
+    Repeated fields in the input are either preserved or discarded except for
+    the last value depending on whether the corresponding schema node is
+    a sequence.
+
+    :type schema: colander.Schema
+    :param params: Query parameter dict, usually `request.params`.
+    :type params: webob.multidict.MultiDict
+    :rtype webob.multidict.MultiDict:
+    :raises ValidationError:
+    """
+    try:
+        # Combine or discard repeated field values depending on type of
+        # corresponding schema node.
+        combined_params = params.dict_of_lists()
+        for key, values in combined_params.items():
+            node = schema.get(key)
+
+            if not node or not isinstance(node.typ, colander.Sequence):
```

Comment:
Is this supposed to be node.typ or is it a typo?

Reply:
`node.typ` is correct, presumably to avoid confusion with the `type` builtin. It would crash if it wasn't.